### ISSUE TITLE
Treeview's disabled list items are not accessible to screen reader users

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#6669]
The experimental Treeview has disabled list items with potentially nested lists inside of them. Visual users can still see these disabled tree nodes and my assumption is that the disabled state _could be_ dependent on some use action elsewhere. If that is the case they should be accessible to screen readers if that's not the case then to quote @carmacleod 
> Why not just remove the branch completely with display:none?

**Comment by dakahn** [comment]
tapping you in on this one @designertyler 🏄

**Comment by dakahn** [comment]
closing in favor of #6792
